File: historyscan-back/src/main/java/io/sch/historyscan/infrastructure/features/analysis/clocrevisions/CodeBaseClocRevisionsFileDTO.java
package io.sch.historyscan.infrastructure.features.analysis.clocrevisions;

import com.fasterxml.jackson.annotation.JsonCreator;
import org.springframework.hateoas.RepresentationModel;

import java.util.Objects;

public class CodeBaseClocRevisionsFileDTO extends RepresentationModel<CodeBaseClocRevisionsFileDTO> {

    private final String name;
    private final String path;
    private final String parentPath;
    private final Long score;

    @JsonCreator
    public CodeBaseClocRevisionsFileDTO(String name, String path, String parentPath, Long score) {
        this.name = name;
        this.path = path;
        this.parentPath = parentPath;
        this.score = score;
    }

    public String getName() {
        return name;
    }

    public String getPath() {
        return path;
    }

    public Long getScore() {
        return score;
    }

    public String getParentPath() {
        return parentPath;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        if (!super.equals(o)) return false;
        CodeBaseClocRevisionsFileDTO that = (CodeBaseClocRevisionsFileDTO) o;
        return Objects.equals(name, that.name)
                && Objects.equals(parentPath, that.parentPath)
                && Objects.equals(path, that.path)
                && Objects.equals(score, that.score);
    }

    @Override
    public int hashCode() {
        return Objects.hash(super.hashCode(), name, parentPath, path, score);
    }
}

File: historyscan-back/src/main/kotlin/io/sch/historyscan/infrastructure/features/analysis/dto/clocrevisions/CodeBaseClocRevisionsFileDTO.kt
package io.sch.historyscan.infrastructure.features.analysis.dto.clocrevisions

import com.fasterxml.jackson.annotation.JsonCreator
import org.springframework.hateoas.RepresentationModel
import java.util.*

class CodeBaseClocRevisionsFileDTO @JsonCreator constructor(val name: String, val path: String, val parentPath: String, val score: Long) : RepresentationModel<CodeBaseClocRevisionsFileDTO?>() {
    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false
        if (!super.equals(o)) return false
        val that = o as CodeBaseClocRevisionsFileDTO
        return name == that.name && parentPath == that.parentPath && path == that.path && score == that.score
    }

    override fun hashCode(): Int {
        return Objects.hash(super.hashCode(), name, parentPath, path, score)
    }
}
